core/services/validation.py

```
@@ -0,0 +1,267 @@
+        
+        # Validate prefix
+        if not message.startswith(allowed_prefix):
+            return 2, f"Message must start with '{allowed_prefix}'"
+        
+        # Extract hash
+        hash_val = message[len(allowed_prefix):]
+        
+        # Lookup in Redis
+        onboarding_data = redis_client.get_active_onboarding(hash_val)
+        if onboarding_data is None:
+            return 2, f"Hash not found or expired"
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)